### DISCUSSION TITLE
Cookies are set when running locally but it does not work when deployed on vercel

### DISCUSSION THREAD

**Discussion by coderpawan** [vercel/next.js#67911] [Help]
### Summary

I have a CreateToken file which gets envoked when the user tries to login and that sets the cookies which is accesed by auth middleware later on for authorising purposes
import jwt from "jsonwebtoken";

const generateToken = (res, userId) => {
  const token = jwt.sign({ userId }, process.env.JWT_SECRET, {
    expiresIn: "30d",
  });

  // Set JWT as an HTTP-Only Cookie

  res.cookie("jwt", token, {
    httpOnly: true,
    secure: false,
    sameSite: "strict",
    maxAge: 30 * 24 * 60 * 60 * 1000,
  });

  return token;
};

export default generateToken;

I could not get where I am getting wrong that it runs locally but not on vercel deployment. I have tried all solutions from different sources but could not resolve. I am stuck on this from last 3 days and is getting irritated like anything. Please help me out

**Comment by icyJoseph** [comment]
Repo example?

**Reply by coderpawan** [reply]
https://github.com/coderpawan/E-commerce

**Reply by icyJoseph** [reply]
Where is the function invoked? it is only exported from the utils, but I don't see where you import it

**Reply by coderpawan** [reply]
> Where is the function invoked? it is only exported from the utils, but I don't see where you import it

It is called as CreateToken function in userController when the user tries to login thereafter I am generating the token and setting it in cookies

**Reply by icyJoseph** [reply]
Aha, apologies, my editor couldn't find the symbol, guess default exports renamed are a bit wacky. I'll take a look now

**Reply by coderpawan** [reply]
> Aha, apologies, my editor couldn't find the symbol, guess default exports renamed are a bit wacky. I'll take a look now

No problem sir but please come up with a solution I am getting disturbed with this error from 3 days and my bad habbit is to stick to the problem until it is solved

**Reply by coderpawan** [reply]
Hey @icyJoseph have you undergone this issue?

**Reply by icyJoseph** [reply]
Hi, sorry I ran out of time. Hopefully today, near the end of the day. Otherwise tomorrow.

**Reply by coderpawan** [reply]
Okk Sir

**Reply by coderpawan** [reply]
Hey @icyJoseph have you figured out something?

**Reply by Kaungsithu-2396** [reply]
@coderpawan is it solved now?
